### PR TITLE
Module/vcf2maf/1.0

### PR DESCRIPTION
**Pull request by whelena**
# Pull Request Checklists

**Important:** When opening a pull request, keep only the applicable checklist and delete all other sections.

### Required

- [ ] I used the cookiecutter template and updated the placeholder rules.

- [x] The snakemake rules follow the [design guidelines](https://lcr-modules.readthedocs.io/en/latest/for_developers.html#module-rules).

- [x] Every rule in the module is either listed under `localrules` or has the `threads` and `resources` directives.

- [ ] Input and output files are being symlinked into the `CFG["inputs"]` and `CFG["outputs"]` subdirectories, respectively.

- [ ] I updated the final target rule (`*_all`) to include every output rule.

- [x] I explained important module design decisions in `CHANGELOG.md`.

- [ ] I tested the module on real data for all supported `seq_type` values.

- [x] I updated the `default.yaml` configuration file to provide default values for each rule in the module snakefile.

- [x] I ensured that all symbolic links are relative and self-contained (_i.e._ do not point outside of the repository).

- [x] I replaced every value that should (or might need to) be updated in the default configuration file with `__UPDATE__`.

- [ ] I recursively searched for all comments containing `TODO` to ensure none were left. For example:

  ```bash
  grep -r TODO modules/<module_name>/1.0
  ```

### If applicable

- [ ] I added more granular output subdirectories.

- [ ] I added rules to the `reference_files` workflow to generate any new reference files.

- [ ] I added subdirectories with large intermediate files to the list of `scratch_subdirectories` in the `default.yaml` configuration file.

- [x] I updated the list of available wildcards for the input files in the `default.yaml` configuration file.

## Checklist for Updated Module

To be completed.

### CODE REVIEW

**Comment by whelena** [comment]
i separate the PR for vcf2maf, this is currently without the auto-installation. I'm wondering if `86-GRCh37` can be added to the local download location at bcgsc.ca to test the auto-installation

File: modules/vcf2maf/1.0/vcf2maf.smk [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+##### SETUP #####
+
+import sys, os
+from os.path import join
+
+import oncopipe as op
+
+# Setup module and store module-specific configuration in `CONFIG`
+CONFIG = config["lcr-modules"]["vcf2maf"]
+LOG = "/logs/" + op._session.launched_fmt
```
Comment (rdmorin) [inline modules/vcf2maf/1.0/vcf2maf.smk:21]: This is different than I'm used to seeing. Can you document what's happening in a comment above this line?

File: modules/vcf2maf/1.0/vcf2maf.smk [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+wildcard_constraints:
```
Comment (rdmorin) [inline modules/vcf2maf/1.0/vcf2maf.smk:23]: Why is this needed? I haven't come across this in other modules

File: modules/vcf2maf/1.0/vcf2maf.smk [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    "GRCh37": "GRCh37",
+    "grch38": "GRCh38",
+    "GRCh38": "GRCh38",
+}
+
+##### RULES #####
+
+rule _vcf2maf_run:
+    input:
+        vcf = "{out_dir}/{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{vcf_name}.vcf",
```
Comment (rdmorin) [inline modules/vcf2maf/1.0/vcf2maf.smk:41]: This will have to handle bgzipped vcfs.

Comment (rdmorin) [reply modules/vcf2maf/1.0/vcf2maf.smk:41]: In fact, let's just force the user to provide bgzipped vcfs!

File: demo/config.yaml [hunk 4]
```diff
@@ -20,3 +20,7 @@ lcr-modules:
         inputs:
             sample_bam: "data/{sample_id}.bam"
             sample_bai: "data/{sample_id}.bam.bai"
+    
+    vcf2maf:
```
[the remark this answers is not in the document]
Comment (whelena) [reply demo/config.yaml:24]: Currently I modified the manta module to support vcf2maf but I can also generate vcf files and put it in bruno's directory

File: modules/manta/2.2/config/default.yaml [hunk 5]
```diff
@@ -0,0 +1,57 @@
+lcr-modules:
```
[the remark this answers is not in the document]
Comment (whelena) [reply modules/manta/2.2/config/default.yaml:1]: Okay I just wanted to see if the vcf2maf module can run seamlessly with other modules

File: modules/vcf2maf/1.0/config/default.yaml [hunk 6]
```diff
@@ -0,0 +1,21 @@
+lcr-modules:
+    
+    vcf2maf:
+        inputs:
+            # Available wildcards: {out_dir}, {seq_type}, {genome_build}, {tumour_id}, {normal_id}, {pair_status}, {vcf_name}
+            vep_cache: "__UPDATE__"
+
+        options:
+            vcf2maf: "--filter-vcf 0 --vcf-tumor-id TUMOR --vcf-normal-id NORMAL"
```
[the remark this answers is not in the document]
Comment (whelena) [reply modules/vcf2maf/1.0/config/default.yaml]: I think varscan uses the same format, I will check with manta

File: modules/vcf2maf/1.0/vcf2maf.smk (hunk 1 above)
Comment (whelena) [reply modules/vcf2maf/1.0/vcf2maf.smk:21]: okay!

File: modules/vcf2maf/1.0/vcf2maf.smk (hunk 2 above)
Comment (whelena) [reply modules/vcf2maf/1.0/vcf2maf.smk:23]: it's because we're not using the `op.setup_module` function. That's because this module is meant to run as an add on to variant callers. So it doesn't have the typical `00-inputs` and `99-outputs` directory. It will simply be generated in a `##-maf` directory within strelka/manta/varscan and be symlinked into their respective `99-outputs/maf/`

File: modules/vcf2maf/1.0/vcf2maf.smk (hunk 3 above)
Comment (whelena) [reply modules/vcf2maf/1.0/vcf2maf.smk:41]: okay! Out of curiosity, do we run vcf2maf on the augmented vcf, whic is not bgzipped (from manta module) or just the direct output, which is already bgzipped?

**Comment by whelena** [comment]
> Leave Star enabled in the demo, yes?

yes just to verify that the new modules doesn't interfere with it. I also haven't tested this module yet and still working on the suggestions you mentioned.

File: modules/vcf2maf/1.0/vcf2maf.smk (hunk 3 above)
Comment (rdmorin) [reply modules/vcf2maf/1.0/vcf2maf.smk:41]: Neither. This would likely never be run on Manta outputs. It's for simple somatic mutations such as the output of strelka and varscan.

File: modules/manta/2.2/config/default.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (rdmorin) [reply modules/manta/2.2/config/default.yaml:1]: yeah, it would be better tested with either Varscan or (maybe better yet) strelka.

[the remark this answers is not in the document]
Comment (whelena) [reply modules/manta/2.2/config/default.yaml:1]: the strelka module is almost finalized, once it is merged to master we can test vcf2maf with that

File: modules/vcf2maf/1.0/vcf2maf.smk (hunk 3 above)
Comment (whelena) [reply modules/vcf2maf/1.0/vcf2maf.smk:41]: oh okayy should we then wait for strelka/varscan to be finalized before testing?

Comment (rdmorin) [reply modules/vcf2maf/1.0/vcf2maf.smk:41]: Almost certainly. If you put Varscan in a separate PR I can help finalize that one but given what I've learned thus far, it will need some changes

Comment (whelena) [reply modules/vcf2maf/1.0/vcf2maf.smk:41]: I have varscan in a separate PR already, but currently it relies on vcf2maf. I can remove vcf2maf from varscan's PR

Comment (rdmorin) [reply modules/vcf2maf/1.0/vcf2maf.smk:41]: sure

File: modules/manta/2.2/config/default.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (rdmorin) [reply modules/manta/2.2/config/default.yaml:1]: Can you look at incorporating it into varscan also? I don't understand how to incorporate it into another pipeline and don't see it in Manta 2.1. Is it in a different branch?

**Comment by rdmorin** [comment]
Is this ready for re-review? @whelena

**Comment by whelena** [comment]
I merged this module with varscan to make testing easier @rdmorin

File: modules/vcf2maf/1.0/config/default.yaml [hunk 7]
```diff
@@ -0,0 +1,23 @@
+lcr-modules:
+    
+    vcf2maf:
+        inputs:
+            # Available wildcards: {out_dir}, {seq_type}, {genome_build}, {tumour_id}, {normal_id}, {pair_status}, {vcf_name}
+            vep_cache: "reference/ve"p_caches/"
```
Comment (rdmorin) [inline modules/vcf2maf/1.0/config/default.yaml:6]: stray quotation mark here?

File: modules/vcf2maf/1.0/config/default.yaml [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+        options:
+            vcf2maf: "--filter-vcf 0 --vcf-tumor-id TUMOR --vcf-normal-id NORMAL --cache-version 97"
+            # --filter-vcf     A VCF for FILTER tag common_variant. Set to 0 to disable [~/.vep/ExAC_nonTCGA.r0.3.1.sites.vep.vcf.gz]
+            #--species        Ensembl-friendly name of species (e.g. mus_musculus for mouse) [homo_sapiens]
+            #--cache-version  Version of offline cache to use with VEP (e.g. 75, 84, 91) [Default: Installed version]
+            species: "homo_sapiens"
+            vep_genome_build: "97_GRCh37"
```
Comment (rdmorin) [inline modules/vcf2maf/1.0/config/default.yaml:14]: Is this hard-coded to only support one of the genome builds?

File: modules/vcf2maf/1.0/config/default.yaml [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+        conda_envs:
+            vcf2maf: "envs/vcf2maf-1.6.18.yaml"
+            
+        threads:
+            vcf2maf: 2
```
Comment (rdmorin) [inline modules/vcf2maf/1.0/config/default.yaml:20]: Increase to 12

**Comment by whelena** [comment]
I'm going to implement the edits on the varscan module/pull request for testing

File: modules/vcf2maf/1.0/config/default.yaml (hunk 8 above)
Comment (whelena) [reply modules/vcf2maf/1.0/config/default.yaml:14]: I added that for auto-installation because 86-GRCh37  wasn't in the gsc link. For vcf2maf-1.0 I think I'll ask the users to provide a path to vep caches. I will keep working on the auto installation for the next versions
